File: esercizi_java/test/compito_python/gestione.py
import sys
import argparse

from cliente import cliente
from Noleggio import Noleggio
from furgone import furgone
from auto import auto


class gestione:
    def __init__(self, arg):
        self.arg = arg

    def main(self):
        veicoli = []
        clienti = []
        codProg = {}

        try:
            f = open("veicoli.txt", "r")
            line = f.readline().strip()
            while line != '':
                tok = line.split()
                cod = int(tok[0])
                tipo = tok[1]
                targa = tok[2]
                line = f.readline().strip()
                modello = line
                line = f.readline().strip()
                marca = line
                line = f.readline().strip()
                if tipo == "auto":
                    tok = line.split()
                    cilindrata = int(tok[0])
                    bagaglio = int(tok[1])
                    line = f.readline().strip()
                    costo = float(line)
                    a = auto(cod, targa, modello, marca,
                             costo, cilindrata, bagaglio)
                    veicoli.append(a)
                    codProg.update({cod: a})

                else:
                    categoria = line
                    line = f.readline().strip()
                    tok = line.split()
                    posti = int(tok[0])
                    vano = bool(tok[1])
                    line = f.readline().strip()
                    costo = float(line)
                    fu = furgone(cod, targa, modello, marca,
                                 costo, categoria, posti, vano)
                    veicoli.append(fu)
                    codProg.update({cod: fu})
                line = f.readline().strip()
            f.close()

        except IOError:
            print("IOError")
        except:
            print("Error")
            raise

        # punto 2

        try:
            f = open("clienti.txt", "r")
            line = f.readline().strip()
            while line != '':
                tok = line.split()
                nome = tok[0]
                cognome = tok[1]
                line = f.readline().strip()

                c = cliente(nome, cognome)
                clienti.append(c)
                tok = line.split()
                x = 0
                while (len(tok) != x):
                    cod = int(tok[x])
                    x = x+1
                    giorni = int(tok[x])
                    x = x+1
                    # chiamata a funzione
                    c.addVeic(codProg.get(cod), giorni)
                line = f.readline().strip()
            f.close()
        except IOError:
            print("IOError")
        except:
            print("Error")
            raise

        # punto 3

        print("tipo, targa, codice, modello, marca, costo giornaliero, cilindrata, bagagliaio, categoria, numero di\n" +
              "posti, vano di carico")
        for v in veicoli:
            print(v.toString())

        #punto 4

        print("\n")
        for c in clienti:
            print(c.toString())


        #punto 5
        print("\n")
        t = 3 #comando dato in arg
        bellissimo = 0
        for c in clienti:
            bellissimo = c.getmassimo(t) + bellissimo
        print(str(bellissimo))



if __name__ == "__main__":
    parser = argparse.ArgumentParser()
    parser.add_argument('matricola', action='store')
    #matr = parser.parse_args()
    matr = 10
    g = gestione(matr)
    g.main()

File: esercizi_java/NetBeansProjects/ciaone/src/gestione/gestione.java
package gestione;
import java.util.List;
import java.util.LinkedList;
import java.util.HashMap;
import java.util.Map;
import java.util.StringTokenizer;
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;


/**
 *
 * @author tobiasacchetto
 */
public class gestione {

    /**
     * @param args the command line arguments
     */
    static List<veicolo> veicoli = new LinkedList<veicolo>();
    static List<cliente> clienti = new LinkedList<cliente>();
    static Map<Integer, veicolo> codProg = new HashMap<Integer, veicolo>();
    
    
    
    public static void main(String[] args) {
        // TODO code application logic here
        
        //punto 1
        
        
        try{
            BufferedReader br = new BufferedReader(new FileReader("veicoli.txt"));
            String line = br.readLine();
            while(line != null){
                StringTokenizer tok = new StringTokenizer(line);
                int cod = Integer.parseInt(tok.nextToken());
                String tipo = tok.nextToken();
                String targa = tok.nextToken();
                line = br.readLine();
                String modello = line;
                line = br.readLine();
                String marca = line;
                line = br.readLine();
                
                if(tipo.equals("auto"))
                {
                    tok = new StringTokenizer(line);
                    int cilindrata = Integer.parseInt(tok.nextToken());
                    int bagaglio = Integer.parseInt(tok.nextToken());
                    line = br.readLine();
                    Double costo = Double.parseDouble(line);
                    
                    auto a = new auto(cod, targa, modello, marca, costo, cilindrata, bagaglio);
                    veicoli.add(a);
                    codProg.put(cod, a);
                    
                }
                else{       
                    String categoria = line;
                    line = br.readLine();
                    tok = new StringTokenizer(line);
                    int posti = Integer.parseInt(tok.nextToken());
                    Boolean vano = Boolean.parseBoolean(tok.nextToken());
                    line = br.readLine();
                    Double costo = Double.parseDouble(line);
                    
                    furgone f = new furgone(cod,targa, modello, marca, costo, categoria, posti, vano);
                    veicoli.add(f);
                    codProg.put(cod,f);
                }
                line = br.readLine();
                
            }
            br.close();
        }catch(IOException e){
        System.err.println(e);
        }catch(Exception e){
        System.err.println(e);
        }
        
        
        
        
        //punto 2
        
        try{
            BufferedReader br = new BufferedReader(new FileReader("clienti.txt"));
            String line = br.readLine();
            
            while(line != null){
            StringTokenizer tok = new StringTokenizer(line);
            String nome = tok.nextToken();
            String cognome = tok.nextToken();
            line = br.readLine();
            tok = new StringTokenizer(line);
            cliente c = new cliente(nome, cognome);
            clienti.add(c);
            while(tok.hasMoreTokens()){
                int cod = Integer.parseInt(tok.nextToken());
                int giorni = Integer.parseInt(tok.nextToken());
                c.addVeic(codProg.get(cod), giorni);
                }
            line = br.readLine();
            }
            br.close();
        }catch(IOException e){
            System.err.println(e);
        }catch(Exception e){
            System.err.println(e);
        }
        
        //punto 3
        
        System.out.println("tipo, targa, codice, modello, marca, costo giornaliero, cilindrata, bagagliaio, categoria, numero di\n" +
"posti, vano di carico");
        for (veicolo v : veicoli){
        System.out.println(v);
        }
        
        
        
        //punto 4
        for(cliente c : clienti){
            System.out.println(c);
        }
        
        
        //punto 5
        System.out.println();
        int veicolo = Integer.parseInt(args[0]);
        double bellissimo=0;
        for(cliente c: clienti)
            bellissimo = c.getmassimo(veicolo)+bellissimo;
        System.out.println(bellissimo);
    }
    
    
}
